File: public/examples/integrations/toolkits/slack/list_conversations_metadata_example_llm_oai.py
import os
from openai import OpenAI

USER_ID = "you@example.com"
PROMPT = "List metadata for all Slack conversations including public channels, private channels, IMs, and group DMs, limiting to 100 results."
TOOL_NAME = "Slack.ListConversationsMetadata"

client = OpenAI(
    base_url="https://api.arcade.dev", api_key=os.environ.get("ARCADE_API_KEY")
)

response = client.chat.completions.create(
    messages=[
        {"role": "user", "content": PROMPT},
    ],
    model="gpt-4o-mini",
    user=USER_ID,
    tools=[TOOL_NAME],
    tool_choice="generate",
)
print(response.choices[0].message.content)

File: public/examples/integrations/toolkits/slack/list_conversations_metadata_example_llm_oai.js
import OpenAI from 'openai';

const USER_ID = 'you@example.com';
const PROMPT = "List metadata for all Slack conversations including public channels, private channels, IMs, and group DMs, limiting to 100 results."
const TOOL_NAME = 'Slack.ListConversationsMetadata';

const client = new OpenAI({
    baseURL: 'https://api.arcade.dev',
    apiKey: process.env.ARCADE_API_KEY
});

const response = await client.chat.completions.create({
    messages: [
        { role: 'user', content: PROMPT }
    ],
    model: 'gpt-4o-mini',
    user: USER_ID,
    tools: [TOOL_NAME],
    tool_choice: 'generate'
});

console.log(response.choices[0].message.content); 